Building docker image fails

I noticed that the `github-docker` worker fails since a couple of days. Building the library exits with the following message:
```
#7 88.70 [38/1042] Building CXX object source/numerics/CMakeFiles/obj_numerics_release.dir/vector_tools_constraints.cc.o
#7 88.70 FAILED: source/numerics/CMakeFiles/obj_numerics_release.dir/vector_tools_constraints.cc.o 
#7 88.70 /usr/bin/c++  -DNDEBUG -Isource/numerics -Iinclude -I../include -isystem /usr/lib/x86_64-linux-gnu/openmpi/include/openmpi -isystem /usr/lib/x86_64-linux-gnu/openmpi/include -isystem /usr/include/petsc -isystem /usr/include/trilinos -isystem /usr/include/hdf5/openmpi -isystem /usr/include/scotch -isystem /usr/include/suitesparse -isystem /usr/include/opencascade -isystem /usr/include/slepc -pedantic -fPIC -Wall -Wextra -Wmissing-braces -Woverloaded-virtual -Wpointer-arith -Wsign-compare -Wsuggest-override -Wswitch -Wsynth -Wwrite-strings -Wno-placement-new -Wno-deprecated-declarations -Wno-literal-suffix -Wno-psabi -Wno-class-memaccess -fopenmp-simd -pthread -std=c++17 -Wno-unused-local-typedefs -O2 -funroll-loops -funroll-all-loops -fstrict-aliasing -MD -MT source/numerics/CMakeFiles/obj_numerics_release.dir/vector_tools_constraints.cc.o -MF source/numerics/CMakeFiles/obj_numerics_release.dir/vector_tools_constraints.cc.o.d -o source/numerics/CMakeFiles/obj_numerics_release.dir/vector_tools_constraints.cc.o -c ../source/numerics/vector_tools_constraints.cc
#7 88.70 In file included from ../include/deal.II/lac/vector_element_access.h:23,
#7 88.70                  from ../include/deal.II/lac/affine_constraints.h:30,
#7 88.70                  from ../include/deal.II/dofs/dof_accessor.templates.h:31,
#7 88.70                  from ../include/deal.II/dofs/dof_accessor.h:2148,
#7 88.70                  from ../include/deal.II/dofs/dof_handler.h:31,
#7 88.70                  from ../include/deal.II/hp/fe_values.h:21,
#7 88.70                  from ../include/deal.II/numerics/vector_tools_constraints.templates.h:22,
#7 88.70                  from ../source/numerics/vector_tools_constraints.cc:17:
#7 88.70 ../include/deal.II/lac/trilinos_tpetra_vector.h:118:11: error: ‘dealii::LinearAlgebra::TpetraWrappers::Vector’ is not a template
#7 88.70   118 |     class Vector : public VectorSpaceVector<Number>, public Subscriptor
#7 88.70       |           ^~~~~~
```

In the [workflow history](https://github.com/dealii/dealii/actions/workflows/docker.yml), I found that it first happened with [this merge commit](https://github.com/dealii/dealii/commit/a483433d327fa8c1a8cd1873ba39506042ffa5b3) related to https://github.com/dealii/dealii/pull/14856.

--- a/include/deal.II/base/template_constraints.h
+++ b/include/deal.II/base/template_constraints.h
@@ -778,28 +778,29 @@ namespace concepts
      * predicate is `true`.
      */
     template <typename T>
-    constexpr bool is_dealii_vector_type = false;
+    inline constexpr bool is_dealii_vector_type = false;
 
     template <typename Number>
-    constexpr bool is_dealii_vector_type<dealii::Vector<Number>> = true;
+    inline constexpr bool is_dealii_vector_type<dealii::Vector<Number>> = true;
 
     template <typename Number>
-    constexpr bool is_dealii_vector_type<dealii::BlockVector<Number>> = true;
+    inline constexpr bool is_dealii_vector_type<dealii::BlockVector<Number>> =
+      true;
 
     template <typename Number>
-    constexpr bool
+    inline constexpr bool
       is_dealii_vector_type<dealii::LinearAlgebra::Vector<Number>> = true;
 
     template <typename Number>
-    constexpr bool
+    inline constexpr bool
       is_dealii_vector_type<dealii::LinearAlgebra::BlockVector<Number>> = true;
 
     template <typename Number, typename MemorySpace>
-    constexpr bool is_dealii_vector_type<
+    inline constexpr bool is_dealii_vector_type<
       dealii::LinearAlgebra::distributed::Vector<Number, MemorySpace>> = true;
 
     template <typename Number>
-    constexpr bool is_dealii_vector_type<
+    inline constexpr bool is_dealii_vector_type<
       dealii::LinearAlgebra::distributed::BlockVector<Number>> = true;
 
 #  ifdef DEAL_II_WITH_PETSC
@@ -836,7 +837,7 @@ namespace concepts
 
 #    ifdef DEAL_II_TRILINOS_WITH_TPETRA
     template <typename Number>
-    constexpr bool is_dealii_vector_type<
+    inline constexpr bool is_dealii_vector_type<
       dealii::LinearAlgebra::TpetraWrappers::Vector<Number>> = true;
 #    endif
 #  endif

--- a/include/deal.II/base/template_constraints.h
+++ b/include/deal.II/base/template_constraints.h
@@ -741,6 +741,7 @@ namespace LinearAlgebra
 #  ifdef DEAL_II_TRILINOS_WITH_TPETRA
   namespace TpetraWrappers
   {
+    template <typename Number>
     class Vector;
   }
 #  endif
@@ -803,40 +804,40 @@ namespace concepts
 
 #  ifdef DEAL_II_WITH_PETSC
     template <>
-    constexpr bool is_dealii_vector_type<dealii::PETScWrappers::Vector> = true;
-
-    template <>
-    constexpr bool is_dealii_vector_type<dealii::PETScWrappers::BlockVector> =
+    inline constexpr bool is_dealii_vector_type<dealii::PETScWrappers::Vector> =
       true;
 
     template <>
-    constexpr bool is_dealii_vector_type<dealii::PETScWrappers::MPI::Vector> =
-      true;
+    inline constexpr bool
+      is_dealii_vector_type<dealii::PETScWrappers::BlockVector> = true;
 
     template <>
-    constexpr bool
+    inline constexpr bool
+      is_dealii_vector_type<dealii::PETScWrappers::MPI::Vector> = true;
+
+    template <>
+    inline constexpr bool
       is_dealii_vector_type<dealii::PETScWrappers::MPI::BlockVector> = true;
 #  endif
 
 #  ifdef DEAL_II_WITH_TRILINOS
     template <>
-    constexpr bool
+    inline constexpr bool
       is_dealii_vector_type<dealii::TrilinosWrappers::MPI::Vector> = true;
 
     template <>
-    constexpr bool
+    inline constexpr bool
       is_dealii_vector_type<dealii::TrilinosWrappers::MPI::BlockVector> = true;
 
     template <>
-    constexpr bool
+    inline constexpr bool
       is_dealii_vector_type<dealii::LinearAlgebra::EpetraWrappers::Vector> =
         true;
 
 #    ifdef DEAL_II_TRILINOS_WITH_TPETRA
-    template <>
-    constexpr bool
-      is_dealii_vector_type<dealii::LinearAlgebra::TpetraWrappers::Vector> =
-        true;
+    template <typename Number>
+    constexpr bool is_dealii_vector_type<
+      dealii::LinearAlgebra::TpetraWrappers::Vector<Number>> = true;
 #    endif
 #  endif
   } // namespace internal

